File: ICPC contests/2008-2009 ACM-ICPC, NEERC, Northern Subregional Contest/H.cpp
// https://codeforces.com/gym/100623

#include <bits/stdc++.h>

using namespace std;

typedef unsigned long long ull;
#define fore(i, a, b) for (ull i = a; i < b; i++)

vector<char> solve(ull h) {
  if (h == 0) {
    return {'1'};
  } else if (h == 1) {
    return {'0'};
  } else if (h % 2 == 0) {
    return vector<char>(h / 2, '8');
  } else {
    vector<char> ans(h / 2 + 1, '8');
    ans[0] = '4';
    return ans;
  }
}

int main(void) {

  // Open file holes.in
  freopen("holes.in", "r", stdin);

  ull h;

  cin >> h;

  // Open file holes.out
  freopen("holes.out", "w", stdout);

  vector<char> ans = solve(h);

  for (char c : ans) {
    cout << c;
  }
  cout << '\n';

  return EXIT_SUCCESS;
}

File: ICPC contests/NEERC Central Subregional 2016-2017/H.py
# https://codeforces.com/gym/101243

import sys
sys.stdin = open("input.txt", "r")
sys.stdout = open("output.txt", "w")


n = int(input())

ans = 8

for i in range(2, min(n+1,10)):
    ans *= 9

if (n >= 10):
    for i in range(10, n+1):
        ans *= 10

print(ans)
